### DISCUSSION TITLE
Blinky-Arm with NUCLEO-F103RB

### DISCUSSION THREAD

**Discussion by jkraskesis** [eclipse-embed-cdt/eclipse-plugins#520] [Debug Plug-ins Q&A]
Hi,

I created the Blinky-Arm tutorial for my NUCLEO-F103RB. It compiles and I am able to run it with QEMU. The reset button works (it restarts the program when pressed). But there is no led that actually blinks on the picture. Should a led visually blink on the board when running the program?

TIA

Joost

**Comment by ilg-ul** [comment]
If you follow the instructions in the official blinky tutorial, using F4DISCOVERY, does the LEDs visually blink?

**Reply by jkraskesis** [reply]
You are right, I should try the official tutotorial before trying to use it for another board.

**Reply by jkraskesis** [reply]
I followed the instructions using F4DISCOVERY and there the LEDs visually blink.

**Reply by ilg-ul** [reply]
Ok, this means that QEMU is functional on your machine. 

The result is that the LEDs visually blink and also write a message in the QEMU console.

If your application doesn't do this, check the GPIO initialisations, you probably missed something. As reference use the functional project.

**Reply by jkraskesis** [reply]
OK, thanks.  I am using the wizard generated Blinky with no code of myself. So I did not miss anything ;-). But changing 
BLINK_PORT_NUMBER to (0),  BLINK_PIN_NUMBER  to  (5) and BLINK_ACTIVE_LOW  to (0) did the trick.

**Reply by ilg-ul** [reply]
Well, the wizard creates generic projects, it does not ask for a specific board, thus you have to customise some definitions.

There is a comment asking for this, perhaps it is visible enough.
